eventcrypto.go

```diff
@@ -78,11 +78,18 @@ func VerifyEventSignatures(ctx context.Context, e PDU, verifier JSONVerifier, us
 
 		// For invites, the invited server should have signed the event.
 		if membership == spec.Invite {
-			_, serverName, err = SplitID('@', *e.StateKey())
-			if err != nil {
-				return fmt.Errorf("failed to split state key: %w", err)
+			switch e.Version() {
+			case RoomVersionV1, RoomVersionV2, RoomVersionV3, RoomVersionV4, RoomVersionV5,
+				RoomVersionV6, RoomVersionV7, RoomVersionV8, RoomVersionV9, RoomVersionV10:
+				_, serverName, err = SplitID('@', *e.StateKey())
+				if err != nil {
+					return fmt.Errorf("failed to split state key: %w", err)
+				}
+				needed[serverName] = struct{}{}
+			default:
+				// TODO: (pseudoIDs) revisit this logic for event signing
```

Comment:
Specifically the invitee should have signed the event, ergo we need a /make_invite which will sign this.